[FIX] Combat system

DiegoForgan opened the pull request:
### Changes on this Pull Request:

* Enemies, weapons, items and spells stats changed
* Enemy spawn logic balanced
*  ADD ATTACK POSITION UPDATE ON BOSSES

DiegoForgan committed:
Enemies, weapons, items and spells stats changed

juanmg0511 commented:
Ahi estoy jugando con el otro pull mergeado, anda todo muy bien pero en los niveles más difíciles es injugable, tenemos que twekear la cantidad de enemigos tambien!

DiegoForgan committed:
Merge branch 'develop' into CombatTunning

DiegoForgan committed:
new enemy spawning logic to add gameplay balance

DiegoForgan committed:
Modified Attack Points for every entity and adapted algorithms

DiegoForgan committed:
dirty comment

DiegoForgan committed:
fixed mid level boss hurting himself

DiegoForgan committed:
added update attackpoints logic to Bosses

DiegoForgan committed:
fixed dead animation bug

Si te morias y te interrumpian la animacion con otra cosa, nunca mas te podias morir y quedabas atrapado. Ahora se soluciona con este commit

DiegoForgan committed:
Final boss spell casting distance reduced

juanmg0511 reviewed (approved):
Approved
